[doc] update deployment guide (#753)
accuknox/KubeArmor: contribution/development_guide.md

## Development

#####1. Vagrant Environment (Recommended)

>    **Note:** You can skip the steps for the vagrant setup if you're directly compiling KubeArmor on any Linux distribution.
>     Please move on to [here](#2-self-managed-kubernetes) to ensure that the steps to setup K8s are followed so as to resolve any open dependencies.

Reviewer: ```suggestion
   **Note** You can skip the steps for the vagrant setup if you're directly compiling KubeArmor on any Linux distribution.
    Please move on to [here](#2-self-managed-kubernetes) to ensure that the steps to setup K8s on the same host are followed so as to resolve any open dependencies.
```

> **Note** Github is adding support to stylize/highlight `Note` in blockqoutes, let's leverage that 😄 

https://github.com/github-community/community/discussions/16925

>     Please move on to [here](#2-self-managed-kubernetes) to ensure that the steps to setup K8s on the same host are followed so as to resolve any open dependencies.

Reviewer 2: ```suggestion
   > **Note** Skip the steps for the vagrant setup if you're directly compiling KubeArmor on the Linux host.
    Proceed [here](#2-self-managed-kubernetes) to setup K8s on the same host by resolving any dependencies.
```

     LSM - AppArmor
     ```

     KubeArmor is designed for Kubernetes, which means that Kubernetes should be ready in your environment. If Kubernetes is not prepared yet, please refer to [Kubernetes installation guide](self-managed-k8s/README.md).
     KubeArmor requires either Docker or Containerd since it internally uses their APIs. KubeArmor also requires LSMs to operate properly; thus, please make sure that your environment supports LSMs \(at least AppArmor\). Otherwise, KubeArmor will work in Audit-Mode with no container behavior restriction.

        ##### Alternative Setup

>         You can try the following alternative if you face any difficulty in the above Kubernetes (kubeadm) setup.

Reviewer: ```suggestion
        You can try the following alternative if you face any difficulty in the above Kubernetes (kubeadm) setup.
        > **Note** Please make sure to setup the alternative k8s environment on the same host as where the KubeArmor developer env is. 
```

     LSM - AppArmor
     ```

>      KubeArmor is designed for Kubernetes, which means that Kubernetes should be ready in your environment. If Kubernetes is not prepared yet, please refer to [Kubernetes installation guide](self-managed-k8s/README.md).

Reviewer 2: ```suggestion
     KubeArmor is designed for Kubernetes environment. If Kubernetes is not setup yet, please refer to [Kubernetes installation guide](self-managed-k8s/README.md).
```

Reviewer: We should mention about `host-only` dev setup too maybe? Should be simpler in case troubles setting up k8s.

Author: For that we have alternative setup

>      KubeArmor requires either Docker or Containerd since it internally uses their APIs. KubeArmor also requires LSMs to operate properly; thus, please make sure that your environment supports LSMs \(at least AppArmor\). Otherwise, KubeArmor will work in Audit-Mode with no container behavior restriction.

Reviewer 2: ```suggestion
     KubeArmor leverages CRI (Container Runtime Interfaces) APIs and works with Docker or Containerd or CRIO based container runtimes. KubeArmor uses LSMs for policy enforcement; thus, please make sure that your environment supports LSMs \(either AppArmor or bpf-lsm\). Otherwise, KubeArmor will operate in Audit-Mode with no policy "enforcement" support.
```
